app/code/local/Magmodules/Webwinkelconnect/Helper/Data.php

```
@@ -97,7 +98,248 @@ public function isProductReviewInviteEnabled(): bool {
+        }
+
+        return true;
+    }
+
+    public function hasCorrectCredentials(?string $webshop_id, ?string $api_key): bool {
+        if (!trim($webshop_id) || !trim($api_key)) {
+            return false;
+        }
+        if ($webshop_id != $this->getShopId() || !hash_equals($api_key, $this->getApiKey())) {
+            return false;
+        }
         return true;
     }
 
+    public function syncProductReview(array $product_review): array {
+        $connection = Mage::getSingleton('core/resource')->getConnection('core_write');
+        if (!$this->isProductReviewInviteEnabled()) {
+            throw new RuntimeException("Product reviews are disabled");
+        }
+        try {
+            $connection->beginTransaction();
+            $product_id = (int) $product_review['product_review']['product_id'];
+            $parent_id = Mage::getModel('catalog/product_type_configurable')->getParentIdsByChild($product_id);
+            if ($parent_id) {
+                $product_id = $parent_id[0];
+            }
+            if (!Mage::getModel('catalog/product')->load($product_id)->getId()) {
+                return [
+                    'message' => $this->__('Could not find product with ID (%d)', $product_id),
+                    'code' => 404
+                ];
+            }
+            $review = Mage::getModel('review/review');
+            if ($product_review['product_review']['id']) {
+                $review = Mage::getModel('review/review')->load($product_review['product_review']['id']);
+            }
+            if ($product_review['product_review']['id'] && !$review->getId()) {
+                return [
+                    'message' => $this->__('Review with ID %s does not exist in Magento', $product_review['product_review']['id']),
+                    'code' => 404
+                ];
+            }
+            if(!empty($product_review['product_review']['deleted']) && $review->getId()) {
```

With this logic, sending a request with `deleted` but without `id` will actually _create_ a new review?

In this case you should return an error.